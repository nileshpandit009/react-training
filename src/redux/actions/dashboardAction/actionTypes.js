export const DASHBOARD_CONSTANTS = {
  FETCH_POSTS: "FETCH_POSTS",
  FETCH_POSTS_SUCCESS: "FETCH_POSTS_SUCCESS",
  FETCH_POSTS_ERROR: "FETCH_POSTS_ERROR",

  FETCH_POST_DETAILS: "FETCH_POST_DETAILS",
  FETCH_POST_DETAILS_SUCCESS: "FETCH_POST_DETAILS_SUCCESS",
  FETCH_POST_DETAILS_ERROR: "FETCH_POST_DETAILS_ERROR",

  RESET_POSTS: "RESET_POSTS",
  RESET_POST_DETAILS: "RESET_POST_DETAILS",
};
