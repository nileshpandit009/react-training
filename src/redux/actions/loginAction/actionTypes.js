export const LOGIN_CONSTANTS = {
  EMAIL: "EMAIL",
  PASSWORD: "PASSWORD",
  EMAIL_ERR: "EMAIL_ERR",
  PASSWORD_ERR: "PASSWORD_ERR",
  LOGIN_ERR: "LOGIN_ERR",
  RESET_LOGIN: "RESET_LOGIN",
  LOGIN_REQUEST: "LOGIN_REQUEST",
  LOGIN_SUCCESS: "LOGIN_SUCCESS",
  LOGIN_FAILURE: "LOGIN_FAILURE",
};
