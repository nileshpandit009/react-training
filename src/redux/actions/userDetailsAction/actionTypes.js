export const USER_DETAILS_CONSTANTS = {
  USER_DETAILS_REQUEST: "USER_DETAILS_REQUEST",
  USER_DETAILS_SUCCESS: "USER_DETAILS_SUCCESS",
  USER_DETAILS_FAILURE: "USER_DETAILS_FAILURE",
  RESET_USER_DETAILS: "RESET_USER_DETAILS",
};
