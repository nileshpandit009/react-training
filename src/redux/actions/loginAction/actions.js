import { LOGIN_CONSTANTS } from "./actionTypes";

const {
  EMAIL,
  PASSWORD,
  EMAIL_ERR,
  PASSWORD_ERR,
  LOGIN_ERR,
  RESET_LOGIN,
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
} = LOGIN_CONSTANTS;

export const changeEmail = (payload) => ({
  type: EMAIL,
  payload,
});

export const changePassword = (payload) => ({
  type: PASSWORD,
  payload,
});

export const changeEmailErr = (payload) => ({
  type: EMAIL_ERR,
  payload,
});

export const changePasswordErr = (payload) => ({
  type: PASSWORD_ERR,
  payload,
});

export const changeLoginErr = (payload) => ({
  type: LOGIN_ERR,
  payload,
});

export const resetLoginState = () => ({
  type: RESET_LOGIN,
});

export const makeLoginRequest = (payload) => ({
  type: LOGIN_REQUEST,
  payload,
});

export const loginSuccess = (payload) => ({
  type: LOGIN_SUCCESS,
  payload,
});

export const loginFailure = (payload) => ({
  type: LOGIN_FAILURE,
  payload,
});
