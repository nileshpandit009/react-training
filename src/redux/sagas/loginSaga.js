import { call, put, takeEvery } from "@redux-saga/core/effects";
import { doLogin } from "apis/loginApi";
import { loginFailure, loginSuccess } from "../actions/loginAction/actions";
import { LOGIN_CONSTANTS } from "../actions/loginAction/actionTypes";

function* makeLoginRequest(action) {
  const { email, password } = action.payload;
  try {
    const userData = yield call(doLogin, email, password);

    localStorage.setItem(
      "user",
      JSON.stringify({
        id: userData.data.id,
        auth_token: userData.data.auth_token,
      })
    );
    yield put(loginSuccess({ authToken: userData.data.auth_token }));
  } catch (e) {
    localStorage.removeItem("user");
    yield put(loginFailure({ error: e.message }));
  }
}

export function* loginSaga() {
  yield takeEvery(LOGIN_CONSTANTS.LOGIN_REQUEST, makeLoginRequest);
}
