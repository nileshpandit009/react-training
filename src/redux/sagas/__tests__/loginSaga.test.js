import { put, takeEvery } from "@redux-saga/core/effects";
import { loginSaga, makeLoginRequest } from "../loginSaga";
import {
  makeLoginRequest as makeLoginRequestAction,
  loginSuccess,
  loginFailure,
} from "redux/actions/loginAction/actions";
import { LOGIN_CONSTANTS } from "redux/actions/loginAction/actionTypes";

describe("Test login saga", () => {
  let makeLoginRequestGenerator;
  beforeEach(() => {
    const loginPayload = {
      email: "test@example.com",
      password: "test password",
    };
    makeLoginRequestGenerator = makeLoginRequest(
      makeLoginRequestAction(loginPayload)
    );
    const callDescriptor = makeLoginRequestGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it("must call loginSuccess action", () => {
    const response = {
      data: {
        id: 123,
        auth_token: "some auth token",
      },
    };

    const putDescriptor = makeLoginRequestGenerator.next(response).value;
    expect(putDescriptor).toEqual(
      put(loginSuccess({ authToken: response.data.auth_token }))
    );

    expect(makeLoginRequestGenerator.next().done).toBeTruthy();
  });

  it("must call loginFailure action", () => {
    const errResponse = {
      message: "some error message",
    };

    const putDescriptor = makeLoginRequestGenerator.throw(errResponse).value;
    expect(putDescriptor).toEqual(
      put(loginFailure({ error: errResponse.message }))
    );

    expect(makeLoginRequestGenerator.next().done).toBeTruthy();
  });
});

describe("Test login saga watchers", () => {
  let loginSagaItr;
  beforeAll(() => {
    loginSagaItr = loginSaga();
  });

  it("must start task to watch for makeLoginRequest action", () => {
    const takeEveryDescriptor = loginSagaItr.next().value;
    expect(takeEveryDescriptor).toEqual(
      takeEvery(LOGIN_CONSTANTS.LOGIN_REQUEST, makeLoginRequest)
    );
  });
});
