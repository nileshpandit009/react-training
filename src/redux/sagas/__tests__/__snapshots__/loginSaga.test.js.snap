// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test login saga must call loginFailure action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "test@example.com",
      "test password",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`Test login saga must call loginSuccess action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "test@example.com",
      "test password",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
